torch/_inductor/codegen/cpp.py

```diff
@@ -1715,13 +1726,22 @@ def store_reduction(self, name, index, value):
                         ),
                         DeferredLine(
                             name,
-                            f"{_lowp_fp_tmpvar_vec}.store({var} + {cexpr_index(index)}, {self.tiling_factor});",
+                            self.get_vec_store_line(
+                                _lowp_fp_tmpvar_vec, var, index, out_dtype
+                            ),
                         ),
                     ]
                 else:
                     raise AssertionError(
                         f"Unsupported reduction type from {dtype} to {out_dtype}"
                     )
+            else:
+                store_lines = [
+                    DeferredLine(
+                        name,
+                        self.get_vec_store_line(value, var, index, out_dtype),
+                    )
+                ]
```

Comment:
This could be deduplicated with the above, just by unconditionally appending this line after all the casting logic.